File: 16-12-2024_K-th_element_of_two_Arrays.cpp
int kthElement(int* a, int n, int* b, int m, int k) {
    int i = 0, j = 0, pos = 0, store = 0;
    
    while (pos != k && i < n && j < m) {
        if (a[i] < b[j]) {
            i++;
            store = 0;
        } else {
            j++;
            store = 1;
        }
        pos++;
    }
    
    while (pos != k && i < n) {
        i++;
        store = 0;
        pos++;
    }
    
    while (pos != k && j < m) {
        j++;
        store = 1;
        pos++;
    }
    
    return store == 0 ? a[i - 1] : b[j - 1];
}

File: 16-12-2024_K-th_element_of_two_Arrays.py
def kthElement(a, n, b, m, k):
    i = 0
    j = 0
    pos = 0
    store = 0
    
    while pos != k and i < n and j < m:
        if a[i] < b[j]:
            i += 1
            store = 0
        else:
            j += 1
            store = 1
        pos += 1
    
    while pos != k and i < n:
        i += 1
        store = 0
        pos += 1
    
    while pos != k and j < m:
        j += 1
        store = 1
        pos += 1
    
    return a[i - 1] if store == 0 else b[j - 1]
